### PR TITLE
MrXL: testing

### PR DESCRIPTION
**Pull request by anshumanmohan**
I felt bad enough about the incorrect merge that I wanted to prevent such things from happening again. Enter, testing.

I've checked these using the Calyx Docker (not the released package, but by building the Dockerfile in the root directory. I did this with the following incantation, while in `calyx/`:
```
docker build -t anshuman/calyx .
docker run -it anshuman/calyx /bin/bash
```
)

Then pull for updates, change to this branch, change to `frontends/mrxl`. Install turnt with `pip install --user turnt` (TODO: request that the Tutorial Docker start off with Turnt).

Now you can run 
1. `turnt -p --env interpret test/*.mrxl`
2. `turnt -p --env gen_calyx test/*.mrxl` (remove the `-p` and add `--save` if you want...)
3. `turnt --env convert_data test/*.mrxl --save` 
4. `turnt --env simulate test/*.mrxl`

3 and 4 are paired: you must `save` step 3's outputs before running step 4.

Known issue: there's something wrong with the Calyx generated for `add.mrxl`.

```
# turnt -vp --env simulate test/add.mrxl --save
$ fud e --from mrxl add.mrxl --to dat --through verilog -s verilog.data add.data
[fud] ERROR: `/home/calyx/target/debug/calyx -l /home/calyx -b verilog --disable-init' failed:
=====STDERR=====
thread 'main' panicked at 'internal error: entered unreachable code: `empty` statements should have been compiled away. Run `compile-empty` before this pass.', calyx-opt/src/passes/top_down_compile_control.rs:453:34
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

=====STDOUT=====
```

### CODE REVIEW

**Comment by anshumanmohan** [comment]
Requesting a review from @susan-garry before anyone else. Let's talk and fix the `add.mrxl` issue, then we'll open it up to the others!

**Comment by susan-garry** [comment]
The bug with `add.mrxl` has been fixed! (The issue was that I was adding an empty `par` group when there weren't any output registers).

One small note: `turnt -v --env interpret test/<prog>.mrxl` compares its output to `test/<prog>.out`, but currently we don't have any `.out` files. We should probably add these files for now, and in the future maybe extend `fud`'s MrXL stage to convert its output into the same format as the MrXL interpreter's output. Then we could merge the `.out` and `.expect` files.

**Comment by anshumanmohan** [comment]
Ah, my bad. I had the incantations for 1 and 2 wrong: I wanted them to use `-p`, not `-v`. Fixed above. Now there isn't a complaint about a missing output file.

**Comment by susan-garry** [comment]
Indeed! Looks good to me!

**Comment by rachitnigam** [comment]
Can we use runt instead of turnt since everything else in the repo uses it and it comes installed in the docker container anlready already?

**Comment by sampsyo** [comment]
Wonderful! Yes, very good to have testing!! I concur with @rachitnigam's request to use Runt because that's what we are doing in this repo; looks like that's all done too. 👏

**Review by sampsyo — APPROVED** [review]
(approving pending #1485)

**Comment by anshumanmohan** [comment]
In case you wonder: #1485 (which moves from turnt to runt) was merged into #1483, which in turn is proposing to merge into this PR. This PR is basically ready to rock, but I'm waiting for #1483 to land so that we never commit turnt-requiring code to the master branch.

**Comment by rachitnigam** [comment]
Woo! Awesome work @susan-garry and @anshumanmohan. Thank you for the amazing push on this!!

**Pull request merged**
